src/app/Fake.Core.Trace/Trace.fs

```diff
@@ -225,20 +226,25 @@ let traceEndTargetUnsafe name =
 [<System.Obsolete("Consider using traceTarget instead and 'use' to properly call traceEndTask in case of exceptions. To remove this warning use 'traceEndTargetUnsafe'.")>]
 let traceEndTarget name = traceEndTargetUnsafe name
 
+let private optionDescriptionToNullable description =
+    match description with 
+    | Some d -> d 
+    | _ -> null
+
 let traceTarget name description dependencyString =
-    traceStartTargetUnsafe name description dependencyString
+    traceStartTargetUnsafe name (optionDescriptionToNullable description) dependencyString
```

Comment:
But this changes the public API of `traceTarget`. We could argue that `traceTarget` is for internal use only but...

Reply:
We could maybe add an `traceTargetInternal` and add a obsolete attribute stating "this feature is for internal use only". On the other hand others (like Xake for example) might want to use this API... So we should probably consider it public

Reply:
That was my argument 😆 

I was also tempted to put the internal modifier, but then because of modules we can't since targets are not internal to trace.

Reply:
ok i'll revert this then and have the narly logic in targets :(